keychain/models/keychain_backend.py

```diff
@@ -47,7 +54,7 @@ def _prepare_keychain(self):
     @api.multi
     def _get_keychain_account(self):
         self.ensure_one()
-        account = self._get_existing_keychain()
+        account = self.keychain_account_id
         if not account:
             vals = self._prepare_keychain()
             account = self.env['keychain.account'].create(vals)
```

Comment:
@andreparames you should assign the created account to `self.keychain_account_id`